database/sql/V072__auction_solution_orders.sql

```diff
@@ -0,0 +1,36 @@
+-- Table to store all proposed solutions for an auction, received from solvers during competition time.
+-- A single auction can have multiple solutions, and each solution can contain multiple orders.
+-- Order is not allowed to be part of more than one solution within the same auction.
+
+-- `solver` + `solution_id` are unique within an auction.
+
+-- `deadline` is used to determine if the order is part of a winning solution.
+
+CREATE TABLE auction_solution_orders (
+   auction_id bigint NOT NULL,
+   solver bytea NOT NULL,
+   solution_id string NOT NULL,
+   order_uid bytea NOT NULL,
+   -- The block number until which the order should be settled.
+   -- Not NULL for winning orders.
+   deadline bigint,
+
+   -- Order details
```

Comment:
Are these needed? Given we store trades for all orders (jit included), do we need to store this information? Also, I think the order uid commits to those values (so you wouldn't be able to change the limit price for a fixed order uid and it can just be read from the settlement).

Reply:
> do we need to store this information? I think the order uid commits to those values

Ah I see, yes, `order_uid` is actually a checksum for `OrderData`. If the surplus capturing surplus JIT order is promised, there are only three outcomes in circuit breaker:
1. Solution not delivered at all - violation.
2. Solution delivered but it doesn't contain promised order_uid of JIT order - violation
3. Solution delivered with the same order uid - then just compare the executed amounts to check if the prices are >= promised.

@fhenneke will circuit breaker require to calculate the score on promised (non-winning) solutions? If not, then we can remove this order data and get them from `orders` table and `jit_orders` table if needed.

Reply:
Ok, after meeting with @fhenneke we concluded that we need this data for all proposed solutions (which include surplus capturing JIT orders and potentially regular JIT orders in the future) and also it makes sense to store this data to show it on the solver competition endpoint.

Reply:
> Can you explain a bit more detailed why we need this information for jit orders even if there ends up not being a settlement observation?

If we have two solutions, and one of them is declared a winner and allowed to settle, and IF the rewards scheme stays the same (difference between winning score and reference score), then we need all data of a second best solution (which might contain surplus capturing JIT order) available at postprocessing time, so that we could calculate the score of it to use it as a reference score.